### PR TITLE
Add known-bug tests for a few I-unsound issues

### PR DESCRIPTION
**Pull request by gburgessiv**
Just a few commits to push #105107 forward.

r? @jackh726

### CODE REVIEW

**Review by a-hosani — APPROVED** [review]

**Comment by jackh726** [comment]
@bors r- 

These tests aren't in the right place, and they should have comments.

File: tests/ui/specialization/issue-105787.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// check-pass
```
Comment (jackh726) [inline tests/ui/specialization/issue-105787.rs:1]: This isn't specialization. Maybe typeck?

**Review by jackh726 — CHANGES_REQUESTED** [review]
A comment in each test briefly saying these should fail and are currently unsound.

**Comment by Dylan-DPC** [comment]
@gburgessiv any updates on this? thanks

**Comment by JohnCSimon** [comment]
@gburgessiv 

Ping from triage: I'm closing this due to inactivity, Please reopen when you are ready to continue with this. 
Note: if you are going to continue please open the PR **BEFORE** you push to it, else you won't be able to reopen - this is a quirk of github. 
Thanks for your contribution.

@rustbot label: +S-inactive

**Pull request closed without merging**